### DISCUSSION TITLE
Installation script for Debian/Ubuntu

### DISCUSSION THREAD

**Discussion by martinkirch** [savonet/liquidsoap#2962] [Show and tell]
Hello,
Since the regular release of `.deb` packages I've had a few difficulties to install it because `dpkg -i` is not enough to install dependencies... So I made a bash script: https://github.com/martinkirch/showergel/blob/main/installers/liquidsoap.sh

In other words, on a fresh Debian or Ubuntu (on an amd64 or arm64 architecture) you can run:

```
wget https://raw.githubusercontent.com/martinkirch/showergel/main/installers/liquidsoap.sh && chmod +x liquidsoap.sh && ./liquidsoap.sh
```

To install or update your liquidsoap binary.

I can contribute it to the main project through a pull request, should I ? If so, how should it be documented ?

Martin

PS: there's a similar script that install Liquidsoap + Showergel, of course ! see https://github.com/martinkirch/showergel#quick-install

**Comment by vitoyucepi** [comment]
Hi @martinkirch,
You can use `apt` to install `deb` packages with dependencies.
For example, in dockerfile
https://gitlab.com/vito-containers/liquidsoap/-/blob/3556400288745ba2564d3e24e9de8b633aa6d15e/containers/2.1/Dockerfile

So there's no need to do this
https://github.com/martinkirch/showergel/blob/b887d2302684b3c1cf72c661b8a5f370efe85384/installers/liquidsoap.sh#L70-L72

**Comment by martinkirch** [comment]
Thanks for the suggestion, I patched the scripts.

**Reply by vitoyucepi** [reply]
Maybe you should check if the user is already *root* and ask if they want to `sudo` the whole script instead. Or `doas` or any other.
https://github.com/martinkirch/showergel/blob/90135941b61eacf6813987dcd5a5baf0d26cae37/installers/liquidsoap.sh#L38-L41

I'm personally not a great fan of `which`. I prefer `command -v`
https://github.com/martinkirch/showergel/blob/90135941b61eacf6813987dcd5a5baf0d26cae37/installers/liquidsoap.sh#L32-L36

You're doing a lot of pre-checks. Why won't you create a bunch of functions that will
- gather info
- do pre-checks

I think that direct download links and checksums are must-haves for packages from the internet.
https://github.com/martinkirch/showergel/blob/90135941b61eacf6813987dcd5a5baf0d26cae37/installers/liquidsoap.sh#L63-L68

<details>
<summary>Also shellcheck</summary>

```
In /tmp/liquidsoap.sh line 51:
    local ARCH="$(dpkg --print-architecture)" # amd64, etc.
          ^--^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In /tmp/liquidsoap.sh line 52:
    source /etc/os-release # comes with a few variables : we'll use ID and VERSION_CODENAME
           ^-------------^ SC1091 (info): Not following: /etc/os-release was not specified as input (see shellcheck -x).


In /tmp/liquidsoap.sh line 63:
    local LATEST=$(curl -w '%{redirect_url}' https://github.com/savonet/liquidsoap/releases/latest)
          ^----^ SC2155 (warning): Declare and assign separately to avoid masking return values.


In /tmp/liquidsoap.sh line 67:
    local PACKAGE=$(ls -tr liquidsoap*.deb |tail)
          ^-----^ SC2155 (warning): Declare and assign separately to avoid masking return values.
                    ^--------------------^ SC2012 (info): Use find instead of ls to better handle non-alphanumeric filenames.


In /tmp/liquidsoap.sh line 68:
    printf "\n\n************ downloaded $PACKAGE **************\n"
           ^-- SC2059 (info): Don't use variables in the printf format string. Use printf '..%s..' "$foo".


In /tmp/liquidsoap.sh line 70:
    sudo apt-get install -y --install-recommends ./$PACKAGE
                                                   ^------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
    sudo apt-get install -y --install-recommends ./"$PACKAGE"

For more information:
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
  https://www.shellcheck.net/wiki/SC1091 -- Not following: /etc/os-release wa...
  https://www.shellcheck.net/wiki/SC2012 -- Use find instead of ls to better ...
```

</details>

**Comment by rmens** [comment]
So I basically did the same thing. But with some different approaches. I might get some inspiration from yours @martinkirch 😊

(https://github.com/oszuidwest/liquidsoap-ubuntu/blob/srt/install.sh)

**Comment by gavin2812** [comment]
Hi @martinkirch just wanted to say thank you for compiling that script, I had been trying for a while to install/compile on Debian, I thank you!
